packages/mattermost-components/src/generic_modal/generic_modal.tsx

```diff
@@ -0,0 +1,521 @@
+
+        const headerText = this.props.modalHeaderText && (
+            <div className='GenericModal__header'>
+                <h1 id='genericModalLabel'>
+                    {this.props.modalHeaderText}
+                </h1>
+            </div>
+        );
+
+        return (
+            <StyledModal
+                dialogClassName={classNames('a11y__modal GenericModal', {GenericModal__compassDesign: this.props.compassDesign}, this.props.className)}
+                show={this.state.show}
+                onHide={this.onHide}
+                onExited={this.props.onExited}
+                enforceFocus={this.props.enforceFocus}
+                restoreFocus={true}
+                role='dialog'
+                aria-label={this.props.ariaLabel}
+                aria-labelledby={this.props.ariaLabel ? undefined : 'genericModalLabel'}
+                id={this.props.id}
+                container={this.props.container}
+                backdrop={this.props.backdrop}
+                backdropClassName={this.props.backdropClassName}
+            >
+                <div
+                    onKeyDown={this.onEnterKeyDown}
+                    tabIndex={0}
+                    className='GenericModal__wrapper-enter-key-press-catcher'
+                >
+                    <Modal.Header closeButton={true}>
+                        {this.props.compassDesign && headerText}
+                    </Modal.Header>
+                    <Modal.Body>
+                        {this.props.compassDesign ? (
+                            this.props.errorText && (
+                                <div className='genericModalError'>
+                                    <i className='icon icon-alert-outline'/>
+                                    <span>{this.props.errorText}</span>
+                                </div>
+                            )
+                        ) : (
+                            headerText
+                        )}
+                        <div className='GenericModal__body'>
+                            {this.props.children}
+                        </div>
+                    </Modal.Body>
+                    {(cancelButton || confirmButton) && <Modal.Footer>
+                        {cancelButton}
+                        {confirmButton}
+                    </Modal.Footer>}
+                </div>
+            </StyledModal>
+        );
+    }
+}
+
+const StyledModal = styled(Modal)`
+&&& {
+    .modal-dialog {
+        margin-top: calc(50vh - 240px);
+    }
+
+    .modal-overflow {
+        .modal-body {
+            overflow: visible;
+        }
+    }
+
+    .modal-body {
+        max-height: 100%;
+        padding: 0;
+
+        .form-control {
+            height: 40px;
+            box-sizing: border-box;
+            border: 2px solid var(--button-bg);
+            border-radius: 4px;
+
+            &.has-error {
+                border-color: var(--error-text);
+            }
+        }
+
+        .MaxLengthInput {
```

Comment:
This isn't something we need to do now, but some of this will likely move back to the web app at some point. I think moving it all at once and then putting back webapp-specific pieces is probably our best bet though similar to how I did the types package

Reply:
Yup. I just took everything related to the component.